src/main/java/network/brightspots/rcv/Tabulator.java

```diff
@@ -365,16 +365,26 @@ private void setWinningThreshold(Map<String, BigDecimal> currentRoundCandidateTo
               config.isHareQuotaEnabled()
                   ? config.getNumberOfWinners()
                   : config.getNumberOfWinners() + 1);
-      if (config.isNonIntegerWinningThresholdEnabled()) {
-        // threshold = (votes / (num_winners + 1)) + 10^(-1 * decimalPlacesForVoteArithmetic)
-        BigDecimal augend =
-            config.divide(
-                BigDecimal.ONE, BigDecimal.TEN.pow(config.getDecimalPlacesForVoteArithmetic()));
-        winningThreshold = config.divide(currentRoundTotalVotes, divisor).add(augend);
+      // If we use integers, we shouldn't use any decimal places.
+      // Otherwise we use the amount of decimal places specified by the user
+      int decimals =
+          config.isNonIntegerWinningThresholdEnabled()
+              ? config.getDecimalPlacesForVoteArithmetic()
+              : 0;
+      // Augend is the smallest unit comaptable our rounding.
+      // If we are only using integers, augend is 1
+      // augend = 10^(-1 * decimals)
+      BigDecimal augend =
+          config.divide(
+              BigDecimal.ONE, BigDecimal.TEN.pow(decimals));
+      if (config.isHareQuotaEnabled()) {
+        // Rounding up simulates "greater than or equal to".
+        // threshold = ceiling(votes / num_winners)
+        winningThreshold = currentRoundTotalVotes.divide(divisor, decimals, java.math.RoundingMode.UP);
       } else {
-        // threshold = floor(votes / (num_winners + 1)) + 1
-        winningThreshold =
-            currentRoundTotalVotes.divideToIntegralValue(divisor).add(BigDecimal.ONE);
+        // Rounding down then adding augend simulates "greater than".
+        // threshold = floor(votes / (numwinners + 1)) + augend
+        winningThreshold = currentRoundTotalVotes.divide(divisor, decimals, java.math.RoundingMode.DOWN).add(augend);
```

Comment:
nit: you can use the existing `divide` method in `ContestConfig` here:
```
winningThreshold = config.divide(currentRoundTotalVotes, divisor).add(augend);
```

Though perhaps we should rename that to `divideAndRoundDown`?

Reply:
I like that name better though, interestingly, `multiply` also rounds down... so we'd probably want to change that too. Anyway, probably fine to handle this outside this particular PR.

Reply:
> nit: you can use the existing divide method in ContestConfig here

I cannot use config.divide here, because the rounding precision isn't necessarily the precision specified in the config file. `decimalPlacesForVoteArithmetic` will be respected iff `nonIntegerWinningThreshold` is true. Otherwise, it will be rounded to the nearest integer.

Reply:
oops, right